src/System.Windows.Forms/src/System/Windows/Forms/KeysConverter.cs

```diff
@@ -230,112 +220,124 @@ public int Compare(object? a, object? b)
-                        {
-                            if (asString)
-                            {
-                                terms.Add(keyString);
-                            }
-                            else
-                            {
-                                terms.Add(keyValue);
-                            }
+                // Finally, if the key wasn't in our list, add it to
+                // the end anyway. Here we just pull the key value out
+                // of the enum.
+                if (!foundKey && Enum.IsDefined(typeof(Keys), (int)keyOnly))
+                {
+                    termKeys.Add(keyOnly);
+                }
 
-                            added = true;
-                            foundKey = true;
-                            break;
-                        }
-                    }
+                return termKeys.ToArray();
+            }
 
-                    // Finally, if the key wasn't in our list, add it to
-                    // the end anyway.  Here we just pull the key value out
-                    // of the enum.
-                    if (!foundKey && Enum.IsDefined(typeof(Keys), (int)keyOnly))
-                    {
-                        if (asString)
-                        {
-                            terms.Add(((Enum)keyOnly).ToString());
-                        }
-                        else
-                        {
-                            terms.Add((Enum)keyOnly);
-                        }
-                    }
+            string GetTermsString(Keys key)
+            {
+                bool added = false;
+                StringBuilder termStrings = new(32);
+                Keys modifiers = (key & Keys.Modifiers);
 
-                    if (asString)
+                // First, iterate through and do the modifiers. These are
+                // additive, so we support things like Ctrl + Alt
+                for (int i = 0; i < DisplayOrder.Count; i++)
+                {
+                    string keyString = DisplayOrder[i];
+                    Keys keyValue = _keyNames[keyString];
+                    if (((int)keyValue & (int)modifiers) != 0)
                     {
-                        StringBuilder b = new StringBuilder(32);
-                        foreach (string t in terms)
+                        if (added)
                         {
-                            b.Append(t);
+                            termStrings.Append('+');
                         }
 
-                        return b.ToString();
+                        termStrings.Append(keyString);
+                        added = true;
                     }
-                    else
+                }
+
+                // Now reset and do the key values. Here, we quit if
+                // we find a match.
+                Keys keyOnly = key & Keys.KeyCode;
+                bool foundKey = false;
+
+                if (added && asString)
```

Comment:
Now that this is refactored into its own method I don't think checking `asString` is necessary anymore.
```suggestion
                if (added)
```